test/PoolManager.t.sol

```diff
@@ -135,12 +140,62 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
         emit ModifyLiquidity(
             key.toId(),
             address(modifyLiquidityRouter),
-            REMOVE_LIQ_PARAMS.tickLower,
-            REMOVE_LIQ_PARAMS.tickUpper,
-            REMOVE_LIQ_PARAMS.liquidityDelta
+            REMOVE_LIQUIDITY_PARAMS.tickLower,
+            REMOVE_LIQUIDITY_PARAMS.tickUpper,
+            REMOVE_LIQUIDITY_PARAMS.liquidityDelta
         );
 
-        modifyLiquidityRouter.modifyLiquidity(key, REMOVE_LIQ_PARAMS, ZERO_BYTES);
+        modifyLiquidityRouter.modifyLiquidity(key, REMOVE_LIQUIDITY_PARAMS, ZERO_BYTES);
+    }
+
+    function test_addLiquidity_withFeeTakingHook() public {
+        address hookAddr =
+            address(uint160(Hooks.AFTER_ADD_LIQUIDITY_FLAG | Hooks.AFTER_ADD_LIQUIDITY_RETURNS_DELTA_FLAG));
+        FeeTakingHook impl = new FeeTakingHook(manager);
+        vm.etch(hookAddr, address(impl).code);
+
+        (key,) = initPool(currency0, currency1, IHooks(hookAddr), 100, SQRT_RATIO_1_1, ZERO_BYTES);
+
+        uint256 balanceBefore0 = currency0.balanceOf(address(this));
+        uint256 balanceBefore1 = currency1.balanceOf(address(this));
+        uint256 hookBalanceBefore0 = currency0.balanceOf(hookAddr);
+        uint256 hookBalanceBefore1 = currency1.balanceOf(hookAddr);
+
+        modifyLiquidityRouter.modifyLiquidity(key, LIQUIDITY_PARAMS, ZERO_BYTES);
+
+        uint256 hookGain0 = currency0.balanceOf(hookAddr) - hookBalanceBefore0;
+        uint256 hookGain1 = currency1.balanceOf(hookAddr) - hookBalanceBefore1;
+        uint256 thisLoss0 = balanceBefore0 - currency0.balanceOf(address(this));
+        uint256 thisLoss1 = balanceBefore1 - currency1.balanceOf(address(this));
+
+        // Assert that the hook got 5.43% of the withdrawn liquidity
+        assertEq(hookGain0, (thisLoss0 - hookGain0) * 543 / 10000, "hook amount 0");
```

Comment:
how come this doesn't check hookGain0 == totalTakenIn0 * 543 / 10000 ?

Reply:
lets say the amount of 0 needed to create this position is 100 tokens, and the fee takes 5.43 tokens.
The user's balance will go down by 105.43 (because they have to pay for the position, and for the fee).

`thisLoss0 - hookGain0` calculates the amount that went to the actual pool manager. To then check that `hookGain0` is 5.43% of it.

I'll add another assert to show that `thisLoss0 - hookGain0` is the amount the PM gained